bundles/org.openhab.binding.tibber/src/main/java/org/openhab/binding/tibber/internal/handler/TibberHandler.java

```diff
@@ -267,6 +273,49 @@ public void getURLInput(String url) throws IOException {
         }
     }
 
+    /**
+     * Builds the {@link TimeSeries} that represents the future tibber prices.
+     * As they are valid for one hour every entry is represented by two time series entries.
+     * 
+     * @param today The prices for today
+     * @param tomorrow The prices for tomorrow.
+     * @return The {@link TimeSeries} with future values.
+     */
+    private TimeSeries buildTimeSeries(JsonArray today, JsonArray tomorrow) {
+        final TimeSeries timeSeries = new TimeSeries(TimeSeries.Policy.REPLACE);
+
+        for (JsonElement entry : today) {
+            DateTimeType startsAt = parseDateTime(entry.getAsJsonObject().get("startsAt"));
+            timeSeries.add( //
+                    startsAt.getInstant(), //
+                    new DecimalType(entry.getAsJsonObject().get("total").getAsString()) //
+            );
+            timeSeries.add( //
+                    startsAt.getInstant().plus(1, ChronoUnit.HOURS).minus(1, ChronoUnit.MICROS), //
+                    new DecimalType(entry.getAsJsonObject().get("total").getAsString()) //
+            );
+        }
+
+        for (JsonElement entry : tomorrow) {
+            DateTimeType startsAt = parseDateTime(entry.getAsJsonObject().get("startsAt"));
+            timeSeries.add( //
+                    startsAt.getInstant(), //
+                    new DecimalType(entry.getAsJsonObject().get("total").getAsString()) //
+            );
+            timeSeries.add( //
+                    startsAt.getInstant().plus(1, ChronoUnit.HOURS).minus(1, ChronoUnit.MICROS), //
+                    new DecimalType(entry.getAsJsonObject().get("total").getAsString()) //
+            );
+        }
```

Comment:
Instead of duplicating this code, could you create the `TimeSeries` outside of this method, and call the method for today and tomorrow respectively, providing the created `TimeSeries`?